src/main/Config.cpp

```diff
@@ -1433,6 +1433,17 @@ Config::processConfig(std::shared_ptr<cpptoml::table> t)
             else if (item.first == "DATABASE")
             {
                 DATABASE = SecretValue{readString(item)};
+                auto connectString = DATABASE.value;
```

Comment:
You can use `Database::isSqlite` condition to avoid re-implementing string search. The warning should probably clarify that future versions of core will only support BucketListDB for ledger state with SQLite for misc data, with all other backends deprecated.